src/mcts/search.cc

```diff
@@ -1510,6 +1510,21 @@ void SearchWorker::DoBackupUpdateSingleNode(
     n->FinalizeScoreUpdate(v / (1.0f + params_.GetShortSightedness() * depth),
                            d, m, node_to_process.multivisit);
 
+    // Flatten policy if we've visited the node enough times. Threshold is
+    // powers of 2, so we can efficiently sample based on visits. This is not
+    // exact with multivisits maybe skipping a sample, but this is good enough.
+    const auto visits = n->GetN();
+    const auto threshold = params_.GetPolicyFlattenThreshold();
+    if (threshold && !(visits & threshold)) {
+      const auto num_edges = n->GetNumEdges();
+      if (num_edges > 0) {
+        const auto flat_policy = 1.0f / num_edges;
+        for (const auto& edge : n->Edges()) {
+          edge.edge()->SetP(0.5f * (edge.GetP() + flat_policy));
```

Comment:
@oscardssmith has a neat suggestion of instead of a param controlling the threshold, determine what's a reasonable rate ahead of time, say every 2^13 = 8192 visits, flatten with a weight param instead of hardcoding 0.5 for average.

Reply:
Gradually mixing a flat policy in can also suffer from the problem where an increase in N without any visits to a specific move can lead to a reduced exploration term for that move as in #1150 which might lead to similar stochasticity from the non-monotonicity.